lib/soci/src/core/session.h

```diff
@@ -77,8 +77,8 @@ class SOCI_DECL session
     template <typename T>
     void set_query_transformation(T callback)
     {
-        std::auto_ptr<details::query_transformation_function> qtf(new details::query_transformation<T>(callback));
-        set_query_transformation_(qtf);
+        std::unique_ptr<details::query_transformation_function> qtf(new details::query_transformation<T>(callback));
```

Comment:
the c++ way is `make_unique`

Reply:
in fairness: this hunk is [closer to upstream](https://github.com/SOCI/soci/blob/master/include/soci/session.h#L89), which somewhat astoundingly provides a `using` alias [for their own `cxx_details::auto_ptr`](https://github.com/SOCI/soci/blob/master/include/soci/soci-platform.h#L120-L125) that, of course, _changes semantics_ between `auto_ptr` and `unique_ptr` depending on whether it's post-C++11. WHAT COULD GO WRONG?

Anyway I think for the sake of smaller diffs against upstream, we ought to leave this hunk alone.

Reply:
ok, I guess that library is doomed anyways if constructors throw for whatever reason